### PR TITLE
move to using quick-xml instead of serde-xml-rs

### LINKED ISSUE #1226: SDK uses unmaintained crate `xml-rs`
`cargo deny check` generates an `unmaintained` warning for xml-rs.

Suggested alternative is `quick-xml`:

```
warning[unmaintained]: xml-rs is Unmaintained

= ID: RUSTSEC-2022-0048
    = Advisory: https://rustsec.org/advisories/RUSTSEC-2022-0048
    = xml-rs is a XML parser has open issues around parsing including integer
      overflows / panics that may or may not be an issue with untrusted data.
      Together with these open issues with Unmaintained status xml-rs
      may or may not be suited to parse untrusted data.
      ## Alternatives
      - [quick-xml](https://crates.io/crates/quick-xml)
    = Announcement: https://github.com/netvl/xml-rs/issues
    = Solution: No safe upgrade is available!
    = xml-rs v0.8.4
      └── serde-xml-rs v0.6.0
          ├── azure_core v0.10.0
          │   ├── azure_storage v0.10.0
          │   │   ├── azure_storage_blobs v0.10.0
          │   │   │   └── azure_storage_cli v0.1.0
          │   │   └── azure_storage_cli v0.1.0 (*)
          │   └── azure_storage_blobs v0.10.0 (*)
          ├── azure_storage v0.10.0 (*)
          └── azure_storage_blobs v0.10.0 (*)
```

### PR DESCRIPTION
**Pull request by demoray**
Fixes #1226

### CODE REVIEW

File: sdk/storage_blobs/src/blob/block_with_size_list.rs [hunk 1]
```diff
@@ -42,13 +45,12 @@ pub struct BlockWithSizeList {
...
     pub fn try_from_xml(xml: &str) -> azure_core::Result<Self> {
-        let bl: BlockList =
-            serde_xml_rs::de::from_reader(xml.as_bytes()).map_kind(ErrorKind::DataConversion)?;
+        let bl: BlockList = read_xml_str(xml).map_kind(ErrorKind::DataConversion)?;
 
         let mut lbs = BlockWithSizeList { blocks: Vec::new() };
 
-        if let Some(b) = bl.committed_blocks.and_then(|c| c.block) {
-            for b_val in b {
+        if let Some(commited) = bl.committed_blocks {
```
Comment (johnbatty) [inline sdk/storage_blobs/src/blob/block_with_size_list.rs:52]: naming: `commited` => `committed`

File: sdk/storage_blobs/src/blob/block_with_size_list.rs [hunk 2]
```diff
@@ -60,8 +62,8 @@ impl BlockWithSizeList {
             }
         }
 
-        if let Some(b) = bl.uncommitted_blocks.and_then(|c| c.block) {
-            for b_val in b {
+        if let Some(commited) = bl.uncommitted_blocks {
+            for b_val in commited.block {
```
Comment (johnbatty) [inline sdk/storage_blobs/src/blob/block_with_size_list.rs:66]: naming: `commited` => `committed`

File: sdk/core/Cargo.toml [hunk 3]
```diff
@@ -28,7 +28,7 @@ reqwest = { version = "0.11", features = [
 ], default-features = false, optional = true }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
-serde-xml-rs = { version = "0.6", optional = true }
+quick-xml = {version="0.27", optional=true, features=["serialize", "serde-types"]}
```
Comment (johnbatty) [inline sdk/core/Cargo.toml:31]: nit: spacing around the `=` to be consistent with the rest of the file

**Review by johnbatty — COMMENTED** [review]
Looks really good - particularly the way you have removed the direct xml crate dependencies from the storage crates, consolidating the function into azure_core.

**Review by cataggar — APPROVED** [review]

**Review by johnbatty — APPROVED** [review]
Looks good to me

**Comment by JelteF** [comment]
Would it be possible to create a new release with this change? I'd quite like to not have the `unmaintained` warning in the `cargo deny check` output.

**Comment by demoray** [comment]
I will investigate doing a new release next week.

**Pull request merged**